src/ShorSimulator.cpp

```diff
@@ -151,137 +152,114 @@ std::map<std::string, std::size_t> ShorSimulator::Simulate([[maybe_unused]] unsi
     return {};
 }
 
-std::pair<unsigned int, unsigned int> ShorSimulator::post_processing(const std::string& sample) const {
-    unsigned long long res = 0;
+/**
+ * Post process the result of the simulation, i.e. try to find two non-trivial factors
+ * @tparam Config Configuration for the underlying DD package
+ * @param sample string with the measurement results (consisting of only 0s and 1s)
+ * @return pair of integers with the factors in case of success or pair of zeros in case of errors
+ */
+template<class Config>
+std::pair<unsigned int, unsigned int> ShorSimulator<Config>::post_processing(const std::string& sample) const {
+    std::ostream log{nullptr};
     if (verbose) {
-        std::clog << "measurement: ";
+        log.rdbuf(std::clog.rdbuf());
     }
```

Comment:
Like the approach. Just out of curiosity: Is it somewhat guaranteed that the compiler bypasses all the logging in case `verbose=False`?

Reply:
It looks nicer but it's still a rather hacky solution. 

It's an `std::ostream` with the [`badbit`](https://en.cppreference.com/w/cpp/io/ios_base/iostate#The_badbit) set. This causes anything sent to output to be discarded. I'm not entirely sure this is guaranteed by the standard, as I find nothing on [std::basic_ostream<CharT,Traits>::operator<<](https://en.cppreference.com/w/cpp/io/basic_ostream/operator_ltlt) and [StackOverflow post](https://stackoverflow.com/questions/7818371/printing-to-nowhere-with-ostream) says to take with a grain of salt.

Btw. I don't think the compiler actually bypasses anything, `verbose` is set at runtime  :S

Reply:
Hm. Yeah, still sounds hacky. But I am fine with that at the moment to be honest.

I was hoping that there maybe is a runtime check somewhere that checks if the output stream even accepts stuff or would discard it anyway and then just bypasses everything that would be fed to the stream. Maybe there is 🤷🏽‍♂️